osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+Script for cron that performs the retention policy
+"""
+import datetime
+import logging
+import os
+import shutil
+from datetime import date
+
+logger = logging.getLogger(__name__)
+
+os.environ['DJANGO_SETTINGS_MODULE'] = 'osh.hub.settings'
+
+
+def delete_tasks_results(tasks_to_delete):
+    from kobo.hub import models
+    for task in tasks_to_delete:
+        try:
+            path_to_delete = models.Task.get_task_dir(task.id)
+            shutil.rmtree(path_to_delete)
+        except OSError as e:
+            logger.error("Error deleting tasks folder: %s", e)
+
+
+# In order to identify personal tasks, we are using the owner field. If the owner doesn't contain any / or @, we assume
+# that it is a personal task.
+def identify_personal_task(owner):
+    # We could use regexps here if we want to make this more complex but for this specific case
+    # built-in string methods are enough, and we are more perfomance-oriented.
+    return "/" not in owner and "@" not in owner
+
+
+def main():
+    import django
+    django.setup()
+
+    # Retention policy Task
+    logger.info("Starting Retention Policy Enforcement")
+
+    from kobo.client.constants import FINISHED_STATES
+
+    from osh.hub.scan import models
+    tasks = tasks = models.Task.objects.filter(state__in=FINISHED_STATES)
+
+    # ---------- Section to take parameters from DB
+
+    retention_personal_tasks = models.RetentionPolicySetting.objects.get_days_from_category("PersonalScan")
+    retention_failed_status = models.RetentionPolicySetting.objects.get_days_from_category("FailedStatus")
+
+    # ---------- End Section to take parameters from DB
+
+    eligible_tasks_to_be_deleted = []
+
+    # We identify which tasks are eligible to be deleted
+    today = date.today()
+    for t in tasks:
+        # At the moment, we only have 2 rules for retention settings:
+        # Personal tasks
+        # Failed status
+        calculated_date_personal_tasks = (t.dt_finished + datetime.timedelta(days=retention_personal_tasks)).date()
+        calculated_date_status_failed = (t.dt_finished + datetime.timedelta(days=retention_failed_status)).date()
+
+        if identify_personal_task(t.owner.username) and calculated_date_personal_tasks <= today:
+            # Personal Scans eligible to be deleted
+            eligible_tasks_to_be_deleted.append(t)
```

Comment:
We need to log log which tasks have been deleted on which criteria.  If we discover that some data is unexpectedly missing one day, there should be a way to audit what exactly happened and why.

Reply:
I believe that this can be now implemented with a `ForeignKey` in `TaskResultsRemoval` pointing to `RetentionPolicySetting`.

Reply:
I added it. Thanks for the comment